isthmus/src/main/java/io/substrait/isthmus/SqlExpressionToSubstrait.java

```diff
@@ -0,0 +1,165 @@
+import io.substrait.type.Type;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.calcite.jdbc.CalciteSchema;
+import org.apache.calcite.prepare.CalciteCatalogReader;
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rel.type.RelDataTypeField;
+import org.apache.calcite.rex.RexInputRef;
+import org.apache.calcite.rex.RexNode;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.parser.SqlParseException;
+import org.apache.calcite.sql.parser.SqlParser;
+import org.apache.calcite.sql.validate.SqlValidator;
+import org.apache.calcite.sql2rel.SqlToRelConverter;
+import org.apache.calcite.sql2rel.StandardConvertletTable;
+
+public class SqlExpressionToSubstrait extends SqlConverterBase {
+
+  protected final RexExpressionConverter rexConverter;
+
+  public SqlExpressionToSubstrait() {
+    this(FEATURES_DEFAULT, EXTENSION_COLLECTION);
+  }
+
+  public SqlExpressionToSubstrait(
+      FeatureBoard features, SimpleExtension.ExtensionCollection extensions) {
+    super(features);
+    ScalarFunctionConverter scalarFunctionConverter =
+        new ScalarFunctionConverter(extensions.scalarFunctions(), factory);
+    this.rexConverter = new RexExpressionConverter(scalarFunctionConverter);
+  }
+
+  @Desugar
+  private record Result(
+      SqlValidator validator,
+      CalciteCatalogReader catalogReader,
+      Map<String, RelDataType> nameToTypeMap,
+      Map<String, RexNode> nameToNodeMap) {}
+
+  /**
+   * Converts the given SQL expression string to an {@link io.substrait.proto.ExtendedExpression }
+   *
+   * @param sqlExpression a SQL expression
+   * @param createStatements table creation statements defining fields referenced by the expression
+   * @return a {@link io.substrait.proto.ExtendedExpression }
+   * @throws SqlParseException
+   */
+  public ExtendedExpression convert(String sqlExpression, List<String> createStatements)
+      throws SqlParseException {
+    var result = registerCreateTablesForExtendedExpression(createStatements);
+    return executeInnerSQLExpression(
+        sqlExpression,
+        result.validator(),
+        result.catalogReader(),
+        result.nameToTypeMap(),
+        result.nameToNodeMap());
+  }
+
+  private ExtendedExpression executeInnerSQLExpression(
+      String sqlExpression,
+      SqlValidator validator,
+      CalciteCatalogReader catalogReader,
+      Map<String, RelDataType> nameToTypeMap,
+      Map<String, RexNode> nameToNodeMap)
+      throws SqlParseException {
+    RexNode rexNode =
+        sqlToRexNode(sqlExpression, validator, catalogReader, nameToTypeMap, nameToNodeMap);
+    NamedStruct namedStruct = toNamedStruct(nameToTypeMap);
+
+    ImmutableExpressionReference expressionReference =
+        ImmutableExpressionReference.builder()
+            .expression(rexNode.accept(this.rexConverter))
+            .addOutputNames("new-column")
```

Comment:
@davisusanibar just for the clarity, 
Should this parameter come from the user input or is this a temporary placeholder which would get updated later?

Reply:
Certainly, this is intentionally assumed as a column name by default, but it could be changed in the future to be defined by the user. Let's wait and see how it is used.